### PR TITLE
Issue 74

### PR DESCRIPTION
**Pull request by cranfillkd**
Connect #74 & #127 

- Added Empty State Page 
- Added Tutorial Popup
- Added “If no data” conditional to dashboard pages
- Added “If no data” conditional to utility energy source page
- Added “If no data” conditional to utility energy bill pages

- Updated Header buttons
- Updated Utility Data Page if no facility is selected
- Updated facility table on Account-Management page 
- Updated routing to reflect empty state

- Removed Dev Tools

Known Bugs:

- Page flickering due to *ngIf=“Subscription”.
- “Add new account” does not reset setup wizard appropriately.

### CODE REVIEW

File: src/app/dashboard/empty-state/empty-state.component.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    this.selectedFacilitySub = this.facilityDbService.selectedFacility.subscribe(selectedFacility => {
+      this.selectedFacility = selectedFacility;
+    });
+
+    this.utilityDataSub = this.utilityMeterDbService.facilityMeters.subscribe(utilityMeters => {
+      this.utilityMeters = utilityMeters;
+    });
+  }
+
```
Comment (rmroot) [inline src/app/dashboard/empty-state/empty-state.component.ts:43]: Need to unsubscribe subscriptions on destroy

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -34,6 +35,11 @@ const routes: Routes = [
         path: 'facility-summary', 
         component: FacilityOverviewComponent,
       }
+      // ,
```
Comment (rmroot) [inline src/app/app-routing.module.ts:38]: Remove commented code if not planning on adding this back or referencing later.

File: src/app/account-management/account/account.component.ts [hunk 3]
```diff
@@ -144,9 +144,10 @@ export class AccountComponent implements OnInit {
   }
 
 
+  /*
```
Comment (rmroot) [inline src/app/account-management/account/account.component.ts:147]: Remove commented code if not planning on adding back or referencing later. Keep if so.

File: src/app/account-management/account/edit-facility/edit-facility.component.html [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<h4>Edit Facility</h4>
+<!--<h4>Edit Facility</h4>
```
Comment (rmroot) [inline src/app/account-management/account/edit-facility/edit-facility.component.html:1]: Remove commented code if not planning on adding back or referencing later. Keep if so.

File: src/app/dashboard/empty-state/empty-state.component.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    this.router.navigate(['/facility-management']);
+  }
+
+  addUtilityData() {
+    this.router.navigate(['/utility/energy-consumption']);
+  }
+
+  loadTestData() {
+    this.accountdbService.addTestData();
+    this.facilityDbService.addTestData();
```
Comment (rmroot) [inline src/app/dashboard/empty-state/empty-state.component.ts:66]: Question: Does this have the problem where a facility gets selected that is not in the account that is selected?

File: src/app/dashboard/facility-overview/facility-overview.component.ts [hunk 6]
```diff
@@ -7,9 +10,15 @@ import { Component, OnInit } from '@angular/core';
...
+  
+  constructor(public utilityMeterDataDbService: UtilityMeterDatadbService) { }
 
   ngOnInit(): void {
+    this.utilityMeterDataSub = this.utilityMeterDataDbService.facilityMeterData.subscribe(utilityMeterFacilityData => {
+      this.utilityMeterFacilityData = utilityMeterFacilityData;
+    });
   }
 
   ngOnDestroy() {
```
Comment (rmroot) [inline src/app/dashboard/facility-overview/facility-overview.component.ts:24]: unsubscribe subcription on destory

File: src/app/shared/setup-progress/setup-progress.component.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+  updateProgress() {
+    if(this.setupWizard) {
+      if(this.selectedAccount && !this.selectedFacility && !this.setupWizardComplete) {
+        this.progress = '33%';
+      } else if(this.selectedAccount && this.selectedFacility && !this.setupWizardComplete) {
+        this.progress = '66%';
+      } else if(this.selectedAccount && this.selectedFacility && this.setupWizardComplete) {
+        this.progress = '100%';
+        const self = this;
+        setTimeout(function(){self.createConfetti();}, 1000);
```
Comment (rmroot) [inline src/app/shared/setup-progress/setup-progress.component.ts:80]: I think "function" is creating the scoping issue. Please try,

`setTimeout(() => {
this.createConfetti();
}, 1000)`

**Review by rmroot — CHANGES_REQUESTED** [review]

File: src/app/dashboard/empty-state/empty-state.component.ts (hunk 5 above)
Comment (cranfillkd) [reply src/app/dashboard/empty-state/empty-state.component.ts:66]: Adding this as an issue

**Review by rmroot — APPROVED** [review]

**Pull request merged**